marklogic-data-hub-central/README.md

```diff
@@ -6,27 +6,72 @@ Boot middle tier instead of a Node middle tier
-the UI. Node is still used for building and running the UI. 
-
-To run the UI, first go into the ./marklogic-data-hub-central/ui directory:
-
-    cd marklogic-data-hub-central/ui
-
-The first time you run the UI, and any time after you've made changes or pulled down changes, you'll need to run 
-the following task to pull down all of the UI dependencies:
-
-    npm install
+the UI. Node is still used for building and running the UI. In addition, you'll also need to have a Data Hub instance 
+that you can talk to. Running each of these processes is described below. 
+
+## Running a Data Hub instance
+
+If you're developing locally and not connecting to a Data Hub instance elsewhere - such as hosted by DHS - you'll need 
+to have a Data Hub instance running locally that you can connect to. You'll typically want to build this instance off 
+the branch that you're working on so that it has the latest changes from the develop branch. 
+
+The CONTRIBUTING.md for this repository discusses how to do this in the "Testing changes to the Data Hub Gradle plugin" section. 
+This process is also explained in detail below for convenience when working on Data Hub Central. 
+
+First, run this command from the root directory in this repository to publish the Data Hub Gradle plugin to your local 
+Maven repository (defaults to ~/.m2/repository, but you won't need to access that directly):
+
+    ./gradlew publishToMavenLocal -PskipWeb=
+
+Every time you pull in new changes to this repository, you should run that command to ensure you are working with the 
+latest change to Data Hub. Also, the "skipWeb" property is included to avoid building the QuickStart application in 
+the "./web" directory, which can require a couple minutes to complete. 
+
+Now, you just need to make a build.gradle file that uses this version of the Data Hub Gradle plugin that you published. 
+For example, if you're working on the develop branch for the 5.3.0 release, then when you publish the DH Gradle plugin, 
+it will have a version of 5.3-SNAPSHOT. Below is a bare minimum build.gradle file for using this after you've published it. 
+
+```
+buildscript {
+    repositories {
+        mavenLocal()
+        jcenter()
+    }
+    dependencies {
+        classpath "net.saliman:gradle-properties-plugin:1.5.1"
+        classpath "com.marklogic:ml-data-hub:5.3-SNAPSHOT"
+    }
+}
+apply plugin: "net.saliman.properties"
+apply plugin: "com.marklogic.ml-data-hub"
+```
+
+You can use this to create your own project and [initialize it with Gradle](https://docs.marklogic.com/datahub/projects/create-project-using-gradle.html). 
+You can also use one of the example projects under "./examples" in this repository - most of them are configured to use 
+a locally published DH Gradle plugin if the property "testing" is set to "true". 
+
+After initializing the project - don't forget to set mlUsername and mlPassword in gradle-local.properties - use Gradle 
```

Comment:
"After initializing the project"... I see that there is an initialization command mentioned in the linked docs page. But the linked docs page also goes through setting up a build.gradle file, which we also talk about here. So there's redundancy. The context switch is very difficult. I think it would be good to cover what the user needs to do in the README, which includes running the hubInit command. (Then as a sidenote, mention how to handle the case where we are using an existing example project.)